packages/devtools_shared/lib/src/deeplink/universal_link_settings.dart

```diff
@@ -28,5 +28,5 @@ extension type const UniversalLinkSettings._(Map<String, Object?> _json) {
 
   /// The associated domains of the iOS build of this Flutter project.
   List<String> get associatedDomains =>
-      _json[_kAssociatedDomainsKey] as List<String>;
+      (_json[_kAssociatedDomainsKey] as List).cast<String>();
```

Comment:
You may need to add `toList()` at the end of this if the return type needs to be a pure List and not a CastList